cmd/shipperctl/cmd/chart/app.go

```diff
@@ -0,0 +1,88 @@
+package chart
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/yaml"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/bookingcom/shipper/cmd/shipperctl/config"
+	shipper "github.com/bookingcom/shipper/pkg/apis/shipper/v1alpha1"
+)
+
+var (
+	appName  string
+	fileName string
+
+	renderAppCmd = &cobra.Command{
+		Use:   "app",
+		Short: "render Shipper Charts for an Application",
+		RunE:  renderChartFromApp,
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			if appName == "" && fileName == "" {
+				cmd.Printf("error: must have *one* of the flags: appName, filename\n")
+				os.Exit(1)
+			}
+			if appName != "" && fileName != "" {
+				cmd.Printf("error: must have *one* of the flags: appName, filename\n")
+				os.Exit(1)
+			}
+		},
+	}
+)
+
+func init() {
+	fileFlagName := "filename"
+
+	renderAppCmd.Flags().StringVar(&appName, "appName", "", "The name of an existing application to render chart for")
+	renderAppCmd.Flags().StringVar(&fileName, fileFlagName, "", "An application manifest in the current context to render chart for (e.g. `applicastion.yaml`)")
+	err := renderAppCmd.MarkFlagFilename(fileFlagName, "yaml")
+	if err != nil {
+		renderAppCmd.Printf("warning: could not mark %q for filename yaml autocompletion: %s\n", fileFlagName, err)
+	}
+
+	renderCmd.AddCommand(renderAppCmd)
+}
+
+func renderChartFromApp(cmd *cobra.Command, args []string) error {
+	c, err := newChartRenderConfig()
+	if err != nil {
+		return err
+	}
+	_, shipperClient, err := config.Load(kubeConfigFile, managementClusterContext)
+	if err != nil {
+		return err
+	}
+	var application shipper.Application
+	if appName != "" {
+		applicationP, err := shipperClient.ShipperV1alpha1().Applications(namespace).Get(appName, metav1.GetOptions{})
```

Comment:
what does `applicationP` stands for?

Reply:
pointer to an application object

Reply:
I found the name a bit confusing.

Could we do:
```go
application, err = *shipperClient.ShipperV1alpha1().Applications(namespace).Get(appName, metav1.GetOptions{})
```

or use `applicationPointer`?

Reply:
using `*` doesn't work because the return type of `.Get()` is `(*Application, error)`
So I renamed

Reply:
Oh of course 🤦